sql/src/lib.rs

```diff
@@ -190,9 +219,26 @@ impl QueryBuilder for SQLiteQueryBuilder {
     }
 
     fn finish(self) -> SQLQuery {
+        // We collected string and UUID arguments into separate maps so that we could
+        // dedupe them. Now we need to turn them into rusqlite Values.
+        let mut args = self.args;
+        let string_args = self.string_args.into_iter().map(|(val, arg)| {
+             (arg, Rc::new(rusqlite::types::Value::Text(val.as_ref().clone())))
+        });
+        let uuid_args = self.uuid_args.into_iter().map(|(val, arg)| {
```

Comment:
Might as well make this `bytes_args` and have the caller handle how to get bytes from their data.  We're likely to store more than just UUIDs as bytes.